### PR TITLE
Add new `Performance/SortWithBlock` cop

### PR DESCRIPTION
**Pull request by koic**
## Feature

This cop identifies places where `sort { |a, b| a.foo <=> b.foo }` can be replaced by `sort_by(&:foo)`.

```ruby
% cat test.rb
# frozen_string_literal: true

array = [1, 2, 3]

array.sort { |a, b| a.to_s <=> b.to_s }
```

```sh
% rubocop test.rb
Inspecting 1 file
C

Offenses:

test.rb:5:7: C: Use sort_by(&:to_s) instead of sort { |a, b| a.to_s <=> b.to_s }.
array.sort { |a, b| a.to_s <=> b.to_s }
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
```

## Target Problem

Enumerable#sort_by { ... } faster than Enumerable#sort { ... }.

Show a benchmark below.

```ruby
require 'benchmark'

array = (1..100000).map { rand(100000) }

Benchmark.bm(10) do |b|
  b.report("sort")    { array.sort { |a, b| a.to_s <=> b.to_s } }
  b.report("sort_by") { array.sort_by(&:to_s) }
end
```

produces:

```sh
                 user     system      total        real
sort         0.570000   0.000000   0.570000 (  0.571723)
sort_by      0.130000   0.000000   0.130000 (  0.129793)
```

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Used the same coding conventions as the rest of the project.
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] All tests are passing.
* [x] The new code doesn't generate RuboCop offenses.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.

[1]: http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

### CODE REVIEW

File: lib/rubocop/cop/performance/sort_with_block.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                        format(MSG, method, var_a, var_b,
+                               var_a, method, var_b, method))
+          end
+        end
+
+        def autocorrect(node)
+          send, = *node
+
+          lambda do |corrector|
+            corrector.replace(sort_range(send, node), 'sort_by(&:foo)')
```
Comment (bbatsov) [inline lib/rubocop/cop/performance/sort_with_block.rb]: You hardcoded `&:foo` here.

File: spec/rubocop/cop/performance/sort_with_block_spec.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# encoding: utf-8
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe RuboCop::Cop::Performance::SortWithBlock do
+  subject(:cop) { described_class.new }
+
+  it 'autocorrects array.sort { |a, b| a.foo <=> b.foo }' do
```
Comment (bbatsov) [inline spec/rubocop/cop/performance/sort_with_block_spec.rb]: Typically we also have some standalone tests testing simply that offenses are being registered/not registered. I'd definitely add some negative test cases.

**Comment by koic** [comment]
Thanks for your review. I fixed that item that was pointed out. How does that look?

**Comment by bbatsov** [comment]
It's in a decent shape. Thanks!

**Comment by koic** [comment]
Thanks too!

**Pull request merged**